src/engine/controls/bpmcontrol.cpp

```diff
@@ -15,6 +15,7 @@
 #include "util/duration.h"
 
 namespace {
+constexpr bool BPM_DEBUG = false;
```

Comment:
someone was saying they didn't like this pattern, but I think it's a lot better than a bunch of commented-out debug statements in our code.  I am happy to entertain alternatives

Reply:
`kLogger.trace()` would be an alternative.

Reply:
does that work on a per-module basis? I usually only want to enable debugging for the engine, or master sync, or midi, but not everything at once -- that's too much noise

Reply:
Right now? No, I don't think so. But the loggers are named, so we could (and should!) add that function. Have a look at `src/util/logger.h`.

Reply:
it looks like the QT way of doing this is through filter rules https://doc.qt.io/qt-5/qloggingcategory.html#setFilterRules.

It looks like implementing per-module debugging will be a significant effort.  What should we do for now?  Keep my current workaround, or just comment out all the debugs?

Reply:
replaced these all with trace calls